subscription.go

```diff
@@ -188,16 +190,47 @@ func (r *marathonClient) registerSSESubscription() error {
 					r.debugLog.Printf("registerSSESubscription(): failed to handle event: %v\n", err)
 				}
 			case err := <-stream.Errors:
-				// TODO let the user handle this error instead of logging it here
-				r.debugLog.Printf("registerSSESubscription(): failed to receive event: %v\n", err)
+				r.debugLog.Printf("registerSSESubscription(): received error from SSE stream: %s", err)
+				stream.Close()
+				go r.registerSSESubscription()
```

Comment:
Does this actually need to invoke another goroutine? Couldn't we just pull the `connectToSSE()` part into for loop starting at line 183?